### PR TITLE
Add blog: Let's create some glip bots with python

### PR DESCRIPTION
**Pull request by zxdong262**

### CODE REVIEW

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+title:  "Let's create some glip bots with python"
+date:   2018-12-10 13:31:07 +0800
+categories: blog
+---
+
+![ ](https://github.com/zxdong262/ringcentral-chatbot-factory-py/blob/master/screenshots/wanted.jpg?raw=true)
+
+This is a intro blog about my recent work: [ringcentral-chatbot-python](https://github.com/zxdong262/ringcentral-chatbot-python).
+
+The project is part of our chatbot plan: Make writing a glip chatbot easy and fun. It is not mature enough for public yet, still need more documents and adding more features.
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: Don't have to capitalize after colon (':').
When referring to an thing to do, either use 'to make writing ....' or 'making writing ...'.
I am not sure how to explain this, but 'It is not mature enough for public yet, needing more documents and more features added still.' would sound more appropriate.

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+It do works and is really easy to work with, you do not even need to know any python to get it running and talking. So ringcentral developers, let's create some chatbot with python, and give some feedback.
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: "It does work ..."
"Do" is the verb here, and you need to change to "does" for third person perspective. Verbs that come after "do" or "does" need to be in first person present tense, like 'I do care about quality', 'you did use the keyboard', or 'he does eat chocolates'.

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+Then try talk to the bot in [https://glip-app.devtest.ringcentral.com](https://glip-app.devtest.ringcentral.com), see how it work.
+
+That is it. You can just focus on the bot logic.
+
+If you want to access user data, subscribe to user event, or use different database, just check these examples.
+
+## Example bots
+
+- [date-time-chatbot](https://github.com/zxdong262/ringcentral-date-time-chatbot) : Simple ringcentral chatbot which can tell time/date.
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: Simple RingCentral chatbot 'that' can tell time/date.
Use proper case for company names, especially our own company name!
Usually if it's in one phrase, use 'that', only use 'which' if you are breaking down the sentence with comma(',').
For example: I made a bot "that" can check my schedule. It is a bot made in python, which is very easy to setup.

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+- [assistant-bot](https://github.com/zxdong262/ringcentral-assistant-bot) : Simple assistant Glip bot to show user/company information, this bot will show you how to access user data.
+- [survey-bot](https://github.com/zxdong262/ringcentral-survey-bot) : Example survey bot, this bot will show you how to create/use custom database wrapper.
+
+## Deploy to AWS Lambda
+
+If you feel ok about the bot, time to deploy to production server, I would suggest AWS Lambda for these reasons:
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: "Feel ok" is too colloquial for writing, you would only use this when you speak.
You can use "If you are happy with the bot", or "If you are satisfied with the bot", or slightly more colloquial and approachable: "If you are OK with the bot".

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+1. Free tier gives 1000000 free run per month
+2. Https ready, do not need extra efforts to handle domain/https staff
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: stuff.
stuff as a noun just means things, staff means 員工.

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+bin/deploy
+
+## watch Lambda server log
+bin/watch
+
+```
+
+- Create API Gateway for your Lambda function, shape as `https://xxxx.execute-api.us-east-1.amazonaws.com/default/poc-your-bot-name-dev-bot/{action+}`
+- Make sure your Lambda function role has permission to read/write dynamodb(Set this from AWS IAM roles, could simply attach `AmazonDynamoDBFullAccess` and `AWSLambdaRole` policies to Lambda function's role)
+- Make sure your Lambda function's timeout more than 5 minutes
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: Make sure your Lambda function's timeout is set to be more than 5 minutes.

File: _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+- Do not forget to set your RingCentral app's redirect URL to Lambda's API Gateway URL, `https://xxxx.execute-api.us-east-1.amazonaws.com/default/poc-your-bot-name-dev-bot/bot-oauth` for bot app.
+
+## Use Extensions
+
+RingCentral Chatbot Framework for Python Extensions will extend bot command support with simple setting in `.env`.
```
Comment (u9520107) [inline _posts/2018-12-10-blog-lets-create some-glip-bots-with-python.markdown]: simple 'settings'.

**Comment by zxdong262** [comment]
@u9520107 Thank you for the detailed review, I should have read more to find more mistakes myself.

**Pull request merged**